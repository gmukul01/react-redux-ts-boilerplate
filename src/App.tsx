import { hot } from 'react-hot-loader/root';
// tslint:disable-next-line: ordered-imports
import React from 'react';
import Button from './components/Button';
import { defaultTheme } from './theme';
import { styled, ThemeProvider } from './utils/styled';

const Text = styled.p`
    color: black;
`;

export const App = () => {
    return (
        <ThemeProvider theme={defaultTheme}>
            <>
                <Text>First Commit</Text>
                <Button solid>Click Button</Button>
            </>
        </ThemeProvider>
    );
};

export const HotApp = hot(App);
