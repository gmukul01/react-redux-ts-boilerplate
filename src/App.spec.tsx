import 'jest-styled-components';
import React from 'react';
import { create } from 'react-test-renderer';
import { App } from './App';

describe('App', () => {
    it('should renders correctly', () => {
        const tree = create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
