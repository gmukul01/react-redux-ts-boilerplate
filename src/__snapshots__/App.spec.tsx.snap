// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
Array [
  .c0 {
  color: black;
}

<p
    className="c0"
  >
    First Commit
  </p>,
  .c0 {
  background-color: transparent;
  border: none;
  padding: 10px;
  margin: 8px;
  border-radius: 3px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #5392f9;
  color: #ffffff;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
    className="c0"
  >
    Click Button
  </button>,
]
`;
